go/private/tools/embed_data.bzl

```diff
@@ -46,17 +54,20 @@ def _go_embed_data_impl(ctx):
   args.add(srcs)
 
   ctx.actions.run(
-      outputs = [ctx.outputs.out],
+      outputs = [out],
       inputs = srcs,
       executable = ctx.executable._embed,
       arguments = [args],
       mnemonic = "GoSourcesData",
   )
+  return [
+      DefaultInfo(files = depset([out])),
+      sources.new(srcs = [out], want_coverage = False),
+  ]
 
 go_embed_data = rule(
     implementation = _go_embed_data_impl,
     attrs = {
-        "out": attr.output(mandatory = True),
```

Comment:
Let's keep this. Removing this attribute will break existing uses.

Also, Gazelle identifies rules with .go files in `out` and `outs` attributes and adds those files to `srcs`. Having this attribute makes it easier to use.

Reply:
It's the only remaining use of attr.output in our rules, it's the wrong way to do things and we ought to stop, mostly because people will copy this rule to write their own code generators.
Having implicit edges by filename (the only reason for the out attribute) is a broken design, and one the Bazel team are trying to steer people away from, if you depend on the outputs of a rule, you should depend on the rule, not a file it happens to produce.

It's not unreasonable to think that we add options like compression to the data embedding rule, and then maybe you want to build sometimes with, and sometimes without, so you want the generated filename to vary with build mode, or you need to add an automatic dependency on the compression library used to go along with the generated source (and thus use a go source to an embed attribute, rather than a file to a src attribute)

I could not find any uses on github right now, and the error produced is fairly obvious to fix (if there are non github users), so I figured a breaking change was acceptable.

Reply:
Fair enough. Gazelle will need to use bazel query to discover generated files, but I think that's something we wanted anyway.